src/main/java/ChatRequest.java

```diff
@@ -0,0 +1,27 @@
+import lombok.*;
+
+import java.util.List;
+
+    /**
+     * @author madhankumar
+     */
+    @Setter
+    @Getter
+    @ToString
+    @Data
+    @Builder
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public class ChatRequest {
+        private String model;
+        private List<?> messages;
+        private int n;
+        private double temperature;
```

Comment:
Using double for temperature may lead to precision issues. Consider using BigDecimal for precise calculations. (Severity: medium)